pkg/finality-grandpa/context.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package grandpa
+
+import (
+	"golang.org/x/exp/constraints"
+)
+
+// The context of a `Round` in which vote weights are calculated.
+type context[ID constraints.Ordered] struct {
+	voters        VoterSet[ID]
+	equivocations bitfield
+}
```

Comment:
Should we consider renaming this struct?
The [context](https://pkg.go.dev/context) package is widely used in Golang. I don't think we should name a struct `context`.

Reply:
This is named this way since there's an analogous type in the rust impl ([link](https://github.com/paritytech/finality-grandpa/blob/5ea95aa79df372457512cff8783022d7124d0329/src/round/context.rs#L30)).  I think we should keep it the same given this is a port of their implementation, and there's a chance we still need to debug if anything goes wrong in future work regarding grandpa.

Reply:
can we name it roundContext then? Naming after standart library pkg is not a best decision. Moreover argument  ` there's an analogous type in the rust impl` is generally false, bcs there is not conclusion that their decisions is better. 
Regarding the debug, we should debug what we have written by understanding what we have written and what is going on there, not by referencing to someones else implementation

Reply:
> can we name it roundContext then? Naming after standart library pkg is not a best decision

I respectfully disagree for the following reasons: 
- This is a private type called `context`, and there is no reference to the standard library `context`.  In my opinion, the only time we would even import the standard `context` library is when we are dealing with requests across API boundaries.  Dave Cheney says that `context.Context` is a [request scoped paradigm](https://dave.cheney.net/2017/08/20/context-isnt-for-cancellation) and should be used as such.  Given that this is just additional context for a GRANDPA round I do think it's a semantically correct name.  I'd prefer it the name over the best practice of just combining the attribute names into a helper struct, aka (`votersEquivocations`).
- It is stored as a private attribute on `Round` and it's not really a good practice to introduce an unnecessary stutter (`Round.roundContext`).  I'll change the type name nevertheless given the amount of push back.  

>Moreover argument  there's an analogous type in the rust impl is generally false, bcs there is not conclusion that their decisions is better.
> Regarding the debug, we should debug what we have written by understanding what we have written and what is going on there, not by referencing to someones else implementation

This package is direct port of parity's reference implementation of the GRANDPA whitepaper.  Given the fact that the actual voter as well validating catch up messages hasn't been fully tested with other clients be it substrate or gossamer based, it's a likely possibility that we will need to debug a running node when testing block authoring capabilities.  It has been extremely helpful to replicate the same tests found in the rust impl, and debugging is much easier when one is a port of another.  This package was ported last year as a learning exercise to determine if it was even possible, and to fulfill functionality (handling concurrent rounds) that is not supported in our current interpretation of the GRANDPA protocol.  We made the decision to try and mirror the GRANDPA client integration as well as the standalone package given how complex it is, and we determined it wasn't feasible to try and adapt our current GRANDPA implementation to work in the same way.